### ISSUE TITLE
Dump error

### ISSUE THREAD

**Issue by toloco** [rethinkdb/rethinkdb#6726]
I am having the following error on rethinkdb dump

**_$> rethinkdb dump_**
```
NOTE: 'rethinkdb-dump' saves data, secondary indexes, and write hooks, but does *not* save
            cluster metadata.  You will need to recreate your cluster setup yourself after
            you run 'rethinkdb-restore'.
  Exporting to temporary directory...
    [                                        ]   0%
    0 rows exported from 1 table, with 0 secondary indexes, and 0 hook functions
Unrecognized TermType: 190.
Errors occurred during export
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/rethinkdb/_dump.py", line 169, in main
    _export.run(options)
  File "/usr/local/lib/python3.7/site-packages/rethinkdb/_export.py", line 524, in run
    run_clients(options, working_dir, db_table_set)
  File "/usr/local/lib/python3.7/site-packages/rethinkdb/_export.py", line 467, in run_clients
    raise RuntimeError("Errors occurred during export")
RuntimeError: Errors occurred during export
Error: export failed, Errors occurred during export
```

I am using a custom version of the official rethinkdb docker images, as it does not contain the python rethinkdb package. I can include if required.

This is the tinniest example I could think of and still failing
```javascript
r.dbCreate("testdb");
r.db("testdb").tableCreate("testtable")
```

**Comment by najamsk** [comment]
wasted 8 hours of my life with my co-worker so 16 hours.  finally on ubuntu 16.0.4 able to at least start export data. but zip file is not creating.



```
sudo rethinkdb dump -f hsnin
NOTE: 'rethinkdb-dump' saves data and secondary indexes, but does *not* save
 cluster metadata.  You will need to recreate your cluster setup yourself after 
 you run 'rethinkdb-restore'.
Exporting to directory...
[========================================] 100% 
3920045 rows exported from 132 tables, with 36 secondary indexes
  Done (161 seconds)
Zipping export directory...
Traceback (most recent call last):
  File "/usr/local/bin/rethinkdb-dump", line 11, in <module>
    load_entry_point('rethinkdb==2.3.0', 'console_scripts', 'rethinkdb-dump')()
  File "/usr/local/lib/python3.5/dist-packages/rethinkdb/_dump.py", line 200, in main
    run_rethinkdb_export(options)
  File "/usr/local/lib/python3.5/dist-packages/rethinkdb/_dump.py", line 183, in run_rethinkdb_export
    do_zip(temp_dir, options)
  File "/usr/local/lib/python3.5/dist-packages/rethinkdb/_dump.py", line 152, in do_zip
    is_fileobj = type(options["out_file"]) is file
NameError: name 'file' is not defined
```


with following pip packages install 

```
pip list
Package             Version               
------------------- ----------------------
pip                 19.0.3                
pycurl              7.43.0                
pygobject           3.20.0                
python-apt          1.1.0b1+ubuntu0.16.4.2
rethinkdb           2.3.0                 
setuptools          40.8.0                
six                 1.12.0                
unattended-upgrades 0.1                   
wheel               0.29.0 
```

**Comment by najamsk** [comment]
on ubuntu 16.0.4 with python3 and python2 both installed. when i choose rethinkdb==2.3.0.post4. it exported files and zip them as well. 

i found following link to help it. 

https://github.com/rethinkdb/rethinkdb/issues/5889

**Comment by codeliger** [comment]
This is still a problem for me on Fedora 29.

`            NOTE: 'rethinkdb-dump' saves data, secondary indexes, and write hooks, but does *not* save
            cluster metadata.  You will need to recreate your cluster setup yourself after
            you run 'rethinkdb-restore'.
  Exporting to temporary directory...
    [                                        ]   0%
    0 rows exported from 2 tables, with 0 secondary indexes, and 0 hook functions
Unrecognized TermType: 190.
Unrecognized TermType: 190.
Errors occurred during export
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/rethinkdb/_dump.py", line 169, in main
    _export.run(options)
  File "/usr/local/lib/python3.7/site-packages/rethinkdb/_export.py", line 524, in run
    run_clients(options, working_dir, db_table_set)
  File "/usr/local/lib/python3.7/site-packages/rethinkdb/_export.py", line 467, in run_clients
    raise RuntimeError("Errors occurred during export")
RuntimeError: Errors occurred during export
Error: export failed, Errors occurred during export`

**Comment by stalem9** [comment]
> on ubuntu 16.0.4 with python3 and python2 both installed. when i choose rethinkdb==2.3.0.post4. it exported files and zip them as well.
> 
> i found following link to help it.
> 
> #5889

Thanks, it works for me on Ubuntu 19 with python3 also

**Comment by marceloabk** [comment]
Same problem with MacOS Mojave 10.14.3

Downgrade to rethinkdb==2.3.0.post4 on python3 solved for me

**Comment by najamsk** [comment]
> > on ubuntu 16.0.4 with python3 and python2 both installed. when i choose rethinkdb==2.3.0.post4. it exported files and zip them as well.
> > i found following link to help it.
> > #5889
> 
> Thanks, it works for me on Ubuntu 19 with python3 also

you welcome

**Comment by c01nd01r** [comment]
Same problem with `rethinkdb-2.4.4.post1`

**Comment by gabor-boros** [comment]
Linking this issue to: https://github.com/rethinkdb/rethinkdb-python/issues/154.

Since I did not see this thread until now,  I only could start to check/fix it today. But doing! 😊

**Comment by gabor-boros** [comment]
Closing the issue, it will be continued within the above-mentioned issue.
In prior: The 2.4.x driver is released for the 2.4.x server.

**Comment by wilywork** [comment]
in version 2.3.6:
* install python 2.7
* set path python in windows
* use pip install rethinkdb==2.3.0
* in your rethinkdb use: rethinkdb dump

Work!
